platforms/chibios/drivers/spi_master.c

```diff
@@ -33,26 +41,42 @@ __attribute__((weak)) void spi_init(void) {
 
         // Try releasing special pins for a short time
         setPinInput(SPI_SCK_PIN);
-        setPinInput(SPI_MOSI_PIN);
-        setPinInput(SPI_MISO_PIN);
+        if (SPI_MOSI_PIN != NO_PIN) {
+            setPinInput(SPI_MOSI_PIN);
+        }
+        if (SPI_MISO_PIN != NO_PIN) {
+            setPinInput(SPI_MISO_PIN);
+        }
 
         chThdSleepMilliseconds(10);
 #if defined(USE_GPIOV1)
         palSetPadMode(PAL_PORT(SPI_SCK_PIN), PAL_PAD(SPI_SCK_PIN), SPI_SCK_PAL_MODE);
-        palSetPadMode(PAL_PORT(SPI_MOSI_PIN), PAL_PAD(SPI_MOSI_PIN), SPI_MOSI_PAL_MODE);
-        palSetPadMode(PAL_PORT(SPI_MISO_PIN), PAL_PAD(SPI_MISO_PIN), SPI_MISO_PAL_MODE);
+        if (SPI_MOSI_PIN != NO_PIN) {
+            palSetPadMode(PAL_PORT(SPI_MOSI_PIN), PAL_PAD(SPI_MOSI_PIN), SPI_MOSI_PAL_MODE);
+        }
+        if (SPI_MISO_PIN != NO_PIN) {
+            palSetPadMode(PAL_PORT(SPI_MISO_PIN), PAL_PAD(SPI_MISO_PIN), SPI_MISO_PAL_MODE);
+        }
 #else
         palSetPadMode(PAL_PORT(SPI_SCK_PIN), PAL_PAD(SPI_SCK_PIN), SPI_SCK_FLAGS);
-        palSetPadMode(PAL_PORT(SPI_MOSI_PIN), PAL_PAD(SPI_MOSI_PIN), SPI_MOSI_FLAGS);
-        palSetPadMode(PAL_PORT(SPI_MISO_PIN), PAL_PAD(SPI_MISO_PIN), SPI_MISO_FLAGS);
+        if (SPI_MOSI_PIN != NO_PIN) {
+            palSetPadMode(PAL_PORT(SPI_MOSI_PIN), PAL_PAD(SPI_MOSI_PIN), SPI_MOSI_FLAGS);
+        }
+        if (SPI_MISO_PIN != NO_PIN) {
+            palSetPadMode(PAL_PORT(SPI_MISO_PIN), PAL_PAD(SPI_MISO_PIN), SPI_MISO_FLAGS);
+        }
 #endif
         spiStop(&SPI_DRIVER);
-        currentSlavePin = NO_PIN;
+        spiStarted = false;
     }
 }
 
 bool spi_start(pin_t slavePin, bool lsbFirst, uint8_t mode, uint16_t divisor) {
-    if (currentSlavePin != NO_PIN || slavePin == NO_PIN) {
+#if SPI_SELECT_MODE == SPI_SELECT_MODE_NONE
+    if (spiStarted) {
+#else
+    if (spiStarted || slavePin == NO_PIN) {
+#endif
         return false;
     }
```

Comment:
Maybe this block breaks `clang-format`?  Then you may try rewriting it in a way that does not have unbalanced braces inside `#if` blocks:
```suggestion
    if (spiStarted) {
        return false;
    }
#if SPI_SELECT_MODE != SPI_SELECT_MODE_NONE
    if (slavePin == NO_PIN) {
        return false;
    }
#endif
```